pkg/controller/status/status_test.go

```diff
@@ -183,6 +186,211 @@ var _ = Describe("Status reporting tests", func() {
 			})
 		})
 
+		Context("when pod is failed", func() {
+			var gen int64
+			BeforeEach(func() {
+				sm.ReadyToMonitor()
+				Expect(client.Create(ctx, &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "NS1",
+						Name:      "DP1pod",
+						Labels: map[string]string{
+							"dp1Key": "dp1Value",
+						},
+					},
+					Spec: corev1.PodSpec{},
+					Status: corev1.PodStatus{
+						Phase: corev1.PodFailed,
+					},
+				})).NotTo(HaveOccurred())
+				Expect(client.Create(ctx, &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "NS1",
```

Comment:
definitely a nit, but the namespace / name values here would be rejected by a real API server for capitalization. Don't think our code cares, though.